_test/shared/matlab_xunit/xunit/TestSuite.m

```diff
@@ -79,14 +79,17 @@
             if nargin < 2
                 monitor = CommandWindowTestRunDisplay();
             end
+            if nargin<3
+                num_tests_runed = 0;
+            end
             
             monitor.testComponentStarted(self);
             did_pass = true;
             
             self.setUp();
             
             for k = 1:numel(self.TestComponents)
-                this_component_passed = self.TestComponents{k}.run(monitor);
+                [this_component_passed,num_tests_runed] = self.TestComponents{k}.run(monitor,num_tests_runed);
```

Comment:
how is `num_tests_runed` being updated by this method?

Reply:
TestComponent, which is acutal executive, adds one to its input value. 
But if you ask how the tests are running in detals -- I do not fully understand it. Its pretty engenious code. This is why Matlab get it for its unit tests (initially it was this code, though Matlab had spoiled it later)